docs/architecture/adr-038-add-support-for-wasm-light-clients.md

```diff
@@ -0,0 +1,77 @@
+# ADR 38: Add support for wasm based light client
+## Changelog
+- {date}: {changelog}
+## Status
+{DRAFT | PROPOSED} Not Implemented
+> Please have a look at the [PROCESS](./PROCESS.md#adr-status) page.
+> Use DRAFT if the ADR is in a draft stage (draft PR) or PROPOSED if it's in review.
+## Abstract
+> "If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the ADR.
+> A short (~200 word) description of the issue being addressed.
+
+Currently in the SDK, light clients are part of cosmos sdk codebase. What this means that, anytime a blockchain built upon cosmos-sdk need to add
+support for new light client, cosmos-sdk need to be modified and all validator nodes of that blockchain need to be updated with custom version of
+cosmos-sdk. 
+
+To remedy these shortcomings, we are proposing a WASM VM to host light client bytecode, which allows easier upgrading of existing light clients
+as well as adding support for new light clients.
+## Context
+> This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply describing facts. It should clearly explain the problem and motivation that the proposal aims to resolve.
+> {context body}
+>
+Currently in the SDK, light clients are defined as part of the codebase and lives at ``. To add support for new light client or
+update existing light client in the event of security issue or consensus update, we need to modify the codebase to add another light client
+module at the location specified and integrate it as light client at other places in the codebase. Apart from this, individual blockchains built upon cosmos-sdk
```

Comment:
... and integrate it _in numerous places_.

Drop "as light client at other places in the codebase".